guardian/grid
# Restrict image download permissions

## What does this change?
[Trello ticket](https://trello.com/c/XY6xdfgt/590-grid-prevents-downloads-for-unpermissioned-users)

In order to lower the risk of picture misuse, this PR restricts making downloading images from Grid behave in the same way as cropping of Paid, Restricted or No rights imagery is available only to permissioned users. Grid will indicate inability to download similarly to inability to crop.

Before
<img width="373" alt="image" src="https://user-images.githubusercontent.com/47318984/182112438-18a9c555-6112-4d0f-bda0-db093580fd33.png">

After
<img width="415" alt="image" src="https://user-images.githubusercontent.com/47318984/182112463-e1fe21ed-f24c-4760-b437-e7be240cfe1c.png">

We have added a config so this feature can be turned on or off. While adding this config, we had to refactor the existing config list to get around the 22 parameters limit in Scala ([context](https://underscore.io/blog/posts/2016/10/11/twenty-two.html#:~:text=Back%20in%202014%2C%20when%20Scala,on%20in%20functions%20and%20tuples.)).


 

## How should a reviewer test this change?

1. Deploy this PR to Grid TEST (unfortunately testing this locally is tricky as you would have to upload an image, which gives you the overriding permission to be able to download the image)
2. Quickest way to test this would be to select an existing image and delete its description (make sure you have the Grid `edit_metadata` rights in Permissions CODE) and refresh the page.
3. The top right corner should say both `Cannot download` and `Cannot crop`



## How can success be measured?
Downloads of non Free to use pictures will be restricted for users without the right permission or if the image has key missing metadata such as credits or description (for reference, a full list of validity checks can be found [here]( https://github.com/guardian/grid/blob/13c1aff4c2362f6a595200883a9015dd0f95b00e/media-api/app/lib/ImageExtras.scala#L15-L23))

## Who should look at this?


## Tested? Documented?
